tools/topology/topology2/cavs-mixin-mixout-efx-hda.conf

```diff
@@ -232,53 +346,74 @@ Object.PCM.pcm [
+                                sink 'gain.1.1'
                         }
                         {
                                 source 'mixin.1.1'
                                 sink 'mixout.2.1'
                         }
                         {
-                                source 'dai-copier.HDA.$HDA_ANALOG_DAI_NAME.capture'
-                                sink 'eqiir.4.1'
-                        }
-                        {
-                                source 'module-copier.4.2'
-                                sink 'host-copier.0.capture'
-                        }
-                        {
-                                source 'host-copier.0.playback'
-                                sink 'gain.1.1'
+                                source 'drc.2.1'
+                                sink 'dai-copier.HDA.$HDA_ANALOG_DAI_NAME.playback'
                         }
                 ]
         }
         "multiband" {
                 Object.Base.route [
                         {
-                                sink 'dai-copier.HDA.$HDA_ANALOG_DAI_NAME.playback'
-                                source 'multiband_drc.2.1'
+                                source 'host-copier.0.playback'
+                                sink 'gain.1.1'
                         }
                         {
                                 source 'mixin.1.1'
                                 sink 'mixout.2.1'
                         }
+                        {
+                                source 'multiband_drc.2.1'
+                                sink 'dai-copier.HDA.$HDA_ANALOG_DAI_NAME.playback'
+                        }
+                ]
+        }
+}
+
+IncludeByKey.HDA_MIC_ENHANCED_CAPTURE {
+       "true" {
+                Object.Base.route [
                         {
                                 source 'dai-copier.HDA.$HDA_ANALOG_DAI_NAME.capture'
                                 sink 'eqiir.4.1'
                         }
                         {
                                 source 'module-copier.4.2'
+                                sink 'tdfb.3.1'
+                        }
+                        {
+                                source 'tdfb.3.1'
+                                sink 'drc.3.1'
+                        }
+                        {
+                                source 'drc.3.1'
                                 sink 'host-copier.0.capture'
                         }
+                ]
+        }
+        "false" {
+                Object.Base.route [
                         {
-                                source 'host-copier.0.playback'
-                                sink 'gain.1.1'
+                                source 'dai-copier.HDA.$HDA_ANALOG_DAI_NAME.capture'
+                                sink 'eqiir.4.1'
+                        }
+                        {
+                                source 'module-copier.4.2'
```

Comment:
e.g. here there's no explicit route from eqiir.4.1 to module-copier.4.2 because it's implicit inside the pipeline number 4?

Reply:
yep